pkg/xds/generator/inbound_proxy_generator.go

```diff
@@ -27,11 +27,17 @@ func (g InboundProxyGenerator) Generate(ctx xds_context.Context, proxy *model.Pr
 	}
 	resources := model.NewResourceSet()
 	for i, endpoint := range endpoints {
+		iface := proxy.Dataplane.Spec.Networking.Inbound[i]
+		protocol := mesh_core.ParseProtocol(iface.GetProtocol())
+
 		// generate CDS resource
 		localClusterName := envoy_names.GetLocalClusterName(endpoint.WorkloadPort)
-		cluster, err := envoy_clusters.NewClusterBuilder().
-			Configure(envoy_clusters.StaticCluster(localClusterName, endpoint.WorkloadIP, endpoint.WorkloadPort)).
-			Build()
+		clusterBuilder := envoy_clusters.NewClusterBuilder().
+			Configure(envoy_clusters.StaticCluster(localClusterName, endpoint.WorkloadIP, endpoint.WorkloadPort))
+		if protocol == mesh_core.ProtocolHTTP2 {
+			clusterBuilder.Configure(envoy_clusters.Http2())
```

Comment:
Should it be like this?
```go
clusterBuilder = clusterBuilder.Configure(envoy_clusters.Http2())
```

Reply:
does not have to be, cluster builder is mutable.